library/core/src/fmt/num.rs

```diff
@@ -339,18 +339,18 @@ macro_rules! impl_Exp {
             // Since `curr` always decreases by the number of digits copied, this means
             // that `curr >= 0`.
             let mut buf = [MaybeUninit::<u8>::uninit(); 40];
-            let mut curr = buf.len() as isize; //index for buf
+            let mut curr = buf.len(); //index for buf
             let buf_ptr = MaybeUninit::slice_as_mut_ptr(&mut buf);
             let lut_ptr = DEC_DIGITS_LUT.as_ptr();
 
             // decode 2 chars at a time
             while n >= 100 {
-                let d1 = ((n % 100) as isize) << 1;
+                let d1 = ((n % 100) << 1) as usize;
```

Comment:
For safety, can you switch this to be more similar to the old one?
```suggestion
                let d1 = ((n % 100) as usize) << 1;
```

The macro defining this function is instantiated with `i8` (https://github.com/rust-lang/rust/blob/3ea44938e21f0de8ae7d4f6399a8a30f97867c70/library/core/src/fmt/num.rs#L465-L468), meaning that `n` could be `i8`, so if `n` was `99_i8` then the shift would overflow if it's done before the cast.

(Can that ever happen?  No idea.  Up to you if you want to try to write a test to trigger it somehow.  But I think it's worth changing the code to obviously not be a problem just in case.)

Reply:
Oh, wait, no, we only get here if `n >= 100`, but `(127_i8 % 100) << 1`, so I guess it'd be fine.

That's still a bit too subtle for my taste, though, so I'd rather just shift in the bigger type.